pages/dashboard/settings.vue

```diff
@@ -4,69 +4,115 @@
-        <input v-model="name" type="text" placeholder="Enter your name" />
-      </label>
-      <h3>Email</h3>
-      <label>
-        <span>
-          The email for your account. This is private information which is not
-          displayed in any API routes or your profile. It is also optional.
-        </span>
-        <input v-model="email" type="email" placeholder="Enter your email" />
-      </label>
-      <h3>Bio</h3>
-      <label>
-        <span>
-          A description of yourself which other users can see on your profile.
-        </span>
-        <input v-model="bio" type="text" placeholder="Enter your bio" />
-      </label>
-    </section>
+    <modal v-show="main_component">
+      <section class="essentials">
+        <h3>Username</h3>
+        <label>
+          <span>
+            The username used on the Modrinth site to identify yourself. This
+            must be unique.
+          </span>
+          <input
+            v-model="username"
+            type="text"
+            placeholder="Enter your username"
+          />
+        </label>
+        <h3>Name</h3>
+        <label>
+          <span>
+            Your display name on your Modrinth profile. This does not have to be
+            unique, can be set to anything, and is optional.
+          </span>
+          <input v-model="name" type="text" placeholder="Enter your name" />
+        </label>
+        <h3>Email</h3>
+        <label>
+          <span>
+            The email for your account. This is private information which is not
+            displayed in any API routes or your profile. It is also optional.
+          </span>
+          <input v-model="email" type="email" placeholder="Enter your email" />
+        </label>
+        <h3>Bio</h3>
+        <label>
+          <span>
+            A description of yourself which other users can see on your profile.
+          </span>
+          <input v-model="bio" type="text" placeholder="Enter your bio" />
+        </label>
+      </section>
+      <section class="essentials pad-maker">
+        <h3>Authorization token</h3>
+        <label>
+          <span>
+            Your authorization token can be used with the Modrinth API and for
+            the Minotaur Gradle plugin. However, it must be kept secret!
+          </span>
+          <button type="button" class="button pad-rem" @click="copyToken">
+            Copy to clipboard
+          </button>
+        </label>
+        <h3>Revoke your token</h3>
+        <label>
+          <span
+            >Beware, this will log you out of Modrinth, and you will have to
+            login again to access Modrinth with a new token.</span
+          >
+          <button type="button" class="button" @click="showDelPopup">
+            Revoke token
+          </button>
+        </label>
+      </section>
+    </modal>
+    <modal v-show="revoke_component">
+      <RevokeTokenHint @wantClose="showDelPopup"></RevokeTokenHint>
+    </modal>
   </DashboardPage>
```

Comment:
This seems like a bad way to this.. Why not just have the revoke hint be on a separate page?

Reply:
Whats the point of it being on another page? Why make the user navigate to a seperate page just to give them 4 sentences and a button that logs them out?

Reply:
Also, this modal system can be extended in the future. I'll be making another PR in the future which will wrap modals allowing for multiple sections. EG: A wizard for modpack creation...